src/cmd/ksh93/tests/head.sh

```diff
@@ -0,0 +1,50 @@
+# Tests for `head` builtin
+
+cat > "$TEST_DIR/file1" <<EOF
+This is line 1 in file1
+This is line 2 in file1
+This is line 3 in file1
+This is line 4 in file1
+This is line 5 in file1
+EOF
+
+cat > "$TEST_DIR/file2" <<EOF2
+This is line 1 in file2
+This is line 2 in file2
+This is line 3 in file2
+This is line 4 in file2
+This is line 5 in file2
+EOF2
+
+# ==========
+#   -n, --lines=lines
+#                   Copy lines lines from each file. The default value is 10.
+actual=$(head -n 3 "$TEST_DIR/file1")
+expect=$'This is line 1 in file1\nThis is line 2 in file1\nThis is line 3 in file1'
+[[ "$actual" = "$expect" ]] || log_error "'head -n' failed"
+
+# ==========
+#   -c, --bytes=chars
+#                   Copy chars bytes from each file.
+actual=$(head -c 23 "$TEST_DIR/file1")
```

Comment:
I would pick a different value so that this verifies you can retrieve part of a line.